src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -887,6 +887,7 @@ private void StepInternal(int threadId, enum_STEPKIND stepKind, SteppingGranular
             // The program should now be stepping, so it is safe to discard the
             // cached program state.
             BeforeContinue();
+            m_isStepping = true;
```

Comment:
>m_isStepping [](http://example.com/codeflow?start=12&length=12)

I am not sure if you want to deal with it, but [https://github.com/microsoft/MIEngine/commit/f91dec6efdf5bd70b06ded5d82301a0968e02b05](https://github.com/microsoft/MIEngine/commit/f91dec6efdf5bd70b06ded5d82301a0968e02b05) also introduces a race here where if we get a stopping event extremely quickly, we could wind up blowing away the stopped state. We really need some sort of lock around the call to step.

Reply:
Tracking at https://github.com/microsoft/MIEngine/issues/1372.